x-pack/plugins/spaces/public/nav_control/solution_view_tour/lib.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, defer, from, of, shareReplay, switchMap } from 'rxjs';
+
+import type { CoreStart } from '@kbn/core/public';
+
+import type { SolutionView } from '../../../common';
+import { SHOW_SPACE_SOLUTION_TOUR_SETTING } from '../../../common/constants';
+import type { SpacesManager } from '../../spaces_manager';
+
+export function initTour(core: CoreStart, spacesManager: SpacesManager) {
+  const showTourUiSettingValue = core.settings.globalClient.get(SHOW_SPACE_SOLUTION_TOUR_SETTING);
+  const hasValueInUiSettings = showTourUiSettingValue !== undefined;
+  const showTour$ = new BehaviorSubject(showTourUiSettingValue ?? true);
+
+  const allSpaces$ = defer(() => from(spacesManager.getSpaces())).pipe(shareReplay(1));
+
+  const hasMultipleSpaces = (spaces: Array<{ solution?: SolutionView }>) => {
+    return spaces.length > 1;
+  };
+
+  const isDefaultSpaceOnClassic = (spaces: Array<{ id: string; solution?: SolutionView }>) => {
+    const defaultSpace = spaces.find((space) => space.id === 'default');
+
+    if (!defaultSpace) {
+      // Don't show the tour if the default space doesn't exist (this should never happen)
+      return true;
+    }
+
+    if (!defaultSpace.solution || defaultSpace.solution === 'classic') {
```

Comment:
> [!TIP]
> We have `SOLUTION_VIEW_CLASSIC` const `classic`

Reply:
When we work with TS types, using constants is not really necessary. I could not have written 'classicc'

Reply:
Addressed in https://github.com/elastic/kibana/pull/194926/commits/91a8d7165316a385d42e131272e1266af1cb9da9